src/BinaryProvider.jl

```diff
@@ -19,13 +19,19 @@ include("Products.jl")
 include("CompatShims.jl")
 
 function __init__()
-    global global_prefix
-
     # Initialize our global_prefix
-    global_prefix = Prefix(joinpath(dirname(pathof(@__MODULE__)), "..", "global_prefix"))
-
+    path = joinpath(@__DIR__, "..", "global_prefix")
```

Comment:
See #156 for my alternative fix. (Though I actually prefer your version personally.)